### ISSUE TITLE
[GKE] Node pool k8s version does not change with the cluster k8s version during cluster create

### ISSUE THREAD

**Issue by aaronRancher** [rancher/rancher#32143]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
From Rancher, navigate to GKE cluster create.
Select a k8s version lower than the default (1.17.17-gke.2800).
Create cluster.

**Result:**
```
Googleapi: Error 400: Node version "1.17.17-gke.2800" must not have a greater minor version than master version "1.16.15-gke.14800"., badRequest
```

**Other details that may be helpful:**
This is because the node pool k8s version does not change when the cluster k8s version changes. This is not a problem when the k8s version of the cluster is higher than the k8s version of the node pool. But this can easily allow the user to create a cluster with a node pool with a higher k8s version than the cluster.
To get around this, the user has to remove the node pool and create a new one in the form. 

As stated by this issue, https://github.com/rancher/rancher/issues/32102, we are not allowing the user to change the node pool k8s version during cluster creation. 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): `v2.5-41f173a-head`
- Installation option (single install/HA): Single Install

**Comment by aaronRancher** [comment]
If the user selects a cluster k8s version greater than 1.17, the node pool version changes to `Upgrade Node Group Version from 1.17.17-gke.2800 to 1.18.15-gke.1502`. The node pool k8s version remains 1.17.17-gke.2800.

**Comment by aaronRancher** [comment]
This bug fix is verified with `v2.5-head` `version 41f173a` and `master-head` `version 1c3629f`

The node pool k8s versions change with the cluster k8s version. The GKE cluster is provisioned with the same k8s version for cluster and node pool.
